### PR TITLE
[jsdocs] Handle directive & schema definitions

### PR DESCRIPTION
**Pull request by phildenhoff**
## Description

See my existing issue for what's up. In short: schema and directives cause the JSDocs plugin to output `[object Object]`, which is kind of annoying. Adding a `DirectiveDefinition` and `SchemaDefinition` which returns `null` solves this problem since the expected behaviour is to not write anything out.

This _does not_ affect directives that are used. If a field is marked `@deprecated(reason: ...)` then that reason will be output as per normal. It just avoids trying to write the definition of the directive to the file.

Related #6505 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots/Sandbox (if appropriate/relevant):

Code sandbox is available here: https://codesandbox.io/s/nifty-euclid-o963o?file=/schema.graphql

## How Has This Been Tested?

1. Include an unused directive in the schema. Nothing extra is printed out, the file looks like it should.
2. Include a deprecated field, marked with the directive. The field is marked as deprecated in the JSDocs as expected.

**Test Environment**:
- OS: macOS
- `@graphql-codegen/jsdoc`: 2.0.0
- NodeJS: v16.1.0

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/plugins/other/jsdoc/src/index.ts [hunk 1]
```diff
@@ -181,6 +187,12 @@ export const plugin: PluginFunction<RawDocumentsConfig> = (schema, documents) =>
         return createDocBlock([createDescriptionBlock(node), `@typedef {*} ${node.name}`]);
       },
     },
+    SchemaDefinition: {
```
Comment (ardatan) [inline packages/plugins/other/jsdoc/src/index.ts:190]: I think this is extra

**Comment by ardatan** [comment]
Could you take a look at my comments and add a changeset so we can merge and release?
Thanks for the PR :)

File: packages/plugins/other/jsdoc/src/index.ts (hunk 1 above)
Comment (phildenhoff) [reply packages/plugins/other/jsdoc/src/index.ts]: Ah you're right! I updated the existing definition to return `null` instead; I found that returning an empty string causes a new line to be generated unnecessarily.

Also: not sure if I created a changeset correctly as I've never used it before. Let me know if I need to change anything!

**Comment by phildenhoff** [comment]
@n1ru4l Tests added! I confirmed they fail without my changes — including avoiding any extra lines generated by `schema {...}`

**Pull request merged**